addons/binding/org.openhab.binding.meterreader/README.md

```diff
@@ -0,0 +1,130 @@
+
+
+## Supported Things
+
+This binding supports only one Thing: `meter`
+
+## Discovery
+
+Discovery is not available, as the binding only reads from serial ports.
+
+## Binding Configuration
+
+No binding configuration required.
+
+## Thing Configuration
+
+The MeterReader thing requires the serial port where the meter device is connected and optionally an refresh intervall.
+
+| Parameter | Name | Description | Required | Default |
+|-----------|------|-------------|----------|---------|
+| `port` | The serial port to connect to| URL to use for playing notification sounds, e.g. `/dev/ttyUSB0` | yes | |
+| `refresh` | The refresh interval in seconds | Defines at which interval the values of the meter device shall be read | no | 20 |
+| `mode` | The protocol mode to use | Can be 'SML' (PUSH mode), 'ABC' (PULL)or D (PUSH) | no | SML |
+| `baudrateChangeDelay` | Delay of baudrate change in ms | USB to serial converters often require a delay of up to 250ms after the ACK before changing baudrate (only relevant for 'C' mode) | no | 0 |
+| `baudrate` | (initial) Baudrate | The baudrate of the serial port. If set to 'auto', it will be negotiated with the meter. The default is 300 baud for modes A, B, and C and 2400 baud for mode D, and 9600 baud for SML. | no | AUTO |
+
+The default is 300 baud for modes A, B, and C and 2400 baud for mode D
+
+## Channels
+
+All available OBIS codes which are read out from the device will be created as channels.
+At every read out the channels will be synched with the OBIS codes from the device.
+
+Following conversion from OBIS codes to channel ID is done:
+`.` is replaced by `-` and `:` or `*` is replaced by `#`.
+
+e.g.
+
+| OBIS code   | Channel ID |
+|-------------|------------|
+|`1-0:1.8.1` | `1-0#1-8-1` |
+|`1.8.0*00` | `1-8-0#00` |
+
+
+### Channel config
+
+**conversionRatio:** This configuration is available for every Number OBIS channels. This is a quotient to easily convert  to a different unit. It is useful whenever the meter provides no/wrong Unit information. Please use default unit conversion provided by openhab.
+For example if the total electrical consumption is given in Wh from the device you usually want to convert it to kWh. To do this use `1000` as conversionRatio.
+
+**negate:** Energy meters often provides absolute values and provide information about the *energy direction* in a separate bit. With this config you can specify the channel where this bit is located, the bit position and the bits value which shall be set.
+
+`<negate> ::= <CHANNEL_ID>:<BIT_POSITION>:<BIT_VALUE>[:status]`
+
+e.g.:
+
+```
+"1-0#1-8-0:5:1:status" // negate if status(1-0#1-8-0) and 2^5 = 1
+"1-0#96-5-5:5:1" // negate if 1-0#96-5-5 and 2^5 = 1
+```
+
+## Unit conversion
+
+Please use the [Units Of Measurement](https://docs.openhab.org/concepts/units-of-measurement.html) concept of openhab for unit conversion which is fully supported by this binding. Please see the item example on how to use it.
```

Comment:
This does not seem to be correct - why else should the user have to deal with "conversion ratios"? And the channel types must be correctly specified with a dimension in that case.

Reply:
A conversionRatio is just in case the meter device wouldn't return a unit. There would be not chance for the user to specify the returned unit manually, right?

Reply:
So what you want to allow here is to specify, which unit the data from the meter actually has, because you cannot identify it yourself? Then the user should set a configuration parameter to "Wh" or "kWh", so that the binding implementation has a chance to create the correct QuantityType. In the item or sitemap file, the user can then still at any time decide to use a different unit in his UIs.

Reply:
Yes, that is a good solution. But I removed it completely now as I currently see now need for it.